pkg/oci/siglayer/siglayer.go

```diff
@@ -28,12 +28,27 @@ import (
 	"github.com/sigstore/sigstore/pkg/cryptoutils"
 )
 
+const (
+	sigkey    = "dev.cosignproject.cosign/signature"
+	certkey   = "dev.sigstore.cosign/certificate"
+	chainkey  = "dev.sigstore.cosign/chain"
+	BundleKey = "dev.sigstore.cosign/bundle"
+)
+
 type sigLayer struct {
 	v1.Layer
-	img  *sigs
+	img  oci.Signatures
 	desc v1.Descriptor
 }
 
+func New(l v1.Layer, img oci.Signatures, desc v1.Descriptor) *sigLayer {
+	return &sigLayer{
+		Layer: l,
+		img:   img,
```

Comment:
Poking around, the only use of `img` seems to be `LayerByDigest` and I wonder if this was before the `v1.Layer` was embedded...  I'd think we could just have `Payload` call `s.Layer.Compressed()` instead of looking it up by the descriptor digest and drop the `img` 🤔

Reply:
that makes sense to me. opened up https://github.com/sigstore/cosign/pull/1042 so we could see if the change makes sense separately!